source/session_encrypt.c

```diff
@@ -168,6 +168,14 @@ static int build_header(struct aws_cryptosdk_session *session, struct aws_crypto
 }
 
 static int sign_header(struct aws_cryptosdk_session *session) {
+    AWS_PRECONDITION(session != NULL);
+    AWS_PRECONDITION(aws_allocator_is_valid(session->alloc));
+    AWS_PRECONDITION(aws_cryptosdk_hdr_is_valid(&session->header));
+    AWS_PRECONDITION(session->header.iv.len <= session->alg_props->iv_len);
+    AWS_PRECONDITION(session->header.auth_tag.len <= session->alg_props->tag_len);
+    AWS_PRECONDITION(aws_cryptosdk_sig_ctx_is_valid(session->signctx));
+    AWS_PRECONDITION(session->state == ST_GEN_KEY);
+    AWS_PRECONDITION(session->mode == AWS_CRYPTOSDK_ENCRYPT);
```

Comment:
We don't need an unique allocator for `session` to cope with proof complexity, but we should have a validity function for it. Could you group the general checks here into a validity function for `aws_cryptosdk_session` structures?

Reply:
Yes! This has been updated!